delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,267 @@
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(id: Iri, project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Update an existing resolver
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param rev            the current revision of the resolver
+    * @param resolverFields the payload to update the resolver
+    */
+  def update(id: Iri, project: ProjectRef, rev: Long, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Deprecate an existing resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    * @param rev     the current revision of the resolver
+    */
+  def deprecate(id: Iri, project: ProjectRef, rev: Long)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Fetch the last version of a resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    */
+  def fetch(id: Iri, project: ProjectRef): UIO[Option[ResolverResource]]
+
+  /**
+    * Fetches the resolver at a given revision
+    * @param id      the resolver identifier
+    * @param project the project where the resolver will belong
+    * @param rev     the current revision of the resolver
+    */
+  def fetchAt(id: Iri, project: ProjectRef, rev: Long): IO[RevisionNotFound, Option[ResolverResource]]
+
+  /**
+    * A non terminating stream of events for resolvers. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def events(offset: Offset = NoOffset): Stream[Task, Envelope[ResolverEvent]]
+
+  /**
+    * The current resolver events. The stream stops after emitting all known events.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def currentEvents(offset: Offset = NoOffset): Stream[Task, Envelope[ResolverEvent]]
+
+}
+
+object Resolvers {
+
+  /**
+    * The resolvers module type.
+    */
+  final val moduleType: String = "resolver"
+
+  private[delta] val inProjectType    = "in-project"
+  private[delta] val crossProjectType = "cross-project"
+
+  import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils.instant
+  import io.scalaland.chimney.dsl._
+
+  private[delta] def next(state: ResolverState, event: ResolverEvent): ResolverState = {
+    (state, event) match {
+      // Creating an in-project resolver
+      case (Initial, e: InProjectResolverCreated)                           =>
+        e.into[InProjectCurrent]
+          .withFieldConst(_.deprecated, false)
```

Comment:
It doesn't seem to bring much better readability than using current directly (`Current(id = ..., deprecated = ...)`)

Reply:
Agree

Reply:
Yep, it is not great, it seems a little more helpful in the evaluate method and not that much either